libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryClient.java

```diff
@@ -71,7 +76,10 @@ private void sendBatch(List<Event> batch, Callback callback) {
       .build();
 
     OkHttpClient client = setting.getClient();
-    client.newCall(request).enqueue(callback);
+
+    if (callback != null) {
```

Comment:
That would mean that if a user doesn't set a callback (`MapboxTelemetry#setHttpCallback`) or if a user sets a callback to `null`, the request won't be executed, which is not valid. Ultimately, registering to a listener should be optional and should not affect the overall behavior ensuring that everything is still working without any action from the client/user.

This could be achieved if `MapboxTelemetry` implements `Callback` and assigning `httpCallback` to `this`, avoiding to change any other classes other than `MapboxTelemetry`.

It'd be great though to add a custom callback which would give us more flexibility on what actions to notify (i.e. to notify actions beyond `onResponse` and `onFailure`), more control over what and when actions need to be notified and it'd open also the possibility of having multiple listeners that users/clients could add/remove as needed.
👀 https://github.com/mapbox/mapbox-java/blob/mas-2.x.x/mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryListener.java
https://github.com/mapbox/mapbox-java/blob/mas-2.x.x/mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java#L656
https://github.com/mapbox/mapbox-java/blob/mas-2.x.x/mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java#L724-L726
https://github.com/mapbox/mapbox-java/blob/mas-2.x.x/mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java#L700-L702
https://github.com/mapbox/mapbox-java/blob/mas-2.x.x/mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java#L78
https://github.com/mapbox/mapbox-java/blob/mas-2.x.x/mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java#L183-L191